pkg/client/unversioned/clientcmd/client_config.go

```diff
@@ -98,7 +98,12 @@ func (config DirectClientConfig) ClientConfig() (*client.Config, error) {
 		u.Fragment = ""
 		clientConfig.Host = u.String()
 	}
-	clientConfig.Version = configClusterInfo.APIVersion
+	// TODO: clientConfig.Version now contains mutiple versions, we'll need to modify related code for version negotiation
```

Comment:
In #14592, I'm making client.Config group specific, so clientConfig.Version will only contain one version. Will that cause any conflicts with this PR?